src/components/dialogs/Dialog.js

```diff
@@ -0,0 +1,113 @@
+import Toggleable from '../../mixins/toggleable'
+
+export default {
+  name: 'dialog',
+
+  mixins: [Toggleable],
+
+  data () {
+    return {
+      stackedActions: false
+    }
+  },
+
+  props: {
+    persistent: Boolean,
+    fullscreen: Boolean,
+    overlay: {
+      type: Boolean,
+      default: true
+    },
+    removeTransition: Boolean,
+    origin: {
+      type: String,
+      default: 'center center'
+    },
+    transition: {
+      type: String,
+      default: 'v-modal-transition'
+    }
+  },
+
+  computed: {
+    classes () {
+      return {
+        'dialog': true,
+        'dialog--active': this.isActive,
+        'dialog--persistent': this.persistent,
+        'dialog--fullscreen': this.fullscreen,
+        'dialog--stacked-actions': this.stackedActions && !this.fullscreen,
+      }
+    },
+  },
+
+  methods: {
+    closeConditional (e) {
+      console.log(e.target)
+      // close dialog if !persistent and clicked outside
+      return this.persistent ? false : true
+    },
+
+    resize () {
+      if (!this.isActive) return
+
+      let actions = this.$children.filter(c => c.$options.propsData.actions)
+
+      // make sure we have the actions card row
+      if (actions.length) {
+        let btns = actions[0].$slots.default
+
+        let maxButtonWidth = (this.$refs.dialog.offsetWidth - 8 - (8 * btns.length)) / btns.length
+        let shouldStack = false
+
+        for (let i=btns.length; i--;) {
+          let span = btns[i].child._vnode.children[0].elm
```

Comment:
$slots array can include empty vnodes (see https://github.com/vuejs/vue/issues/5329). Perhaps include a check if child is defined?